nise/generators/azure/azure_generator.py

```diff
@@ -352,11 +352,13 @@ def _update_data(self, row, start, end, **kwargs):
         if not service_info_2:
             service_info_2 = ""
 
+        _meter_name = self._meter_name if self._meter_name else meter_name
```

Comment:
Should this value be set during initilaization? It doesn't look like there is a way to set this value during initilazation, which is a problem with the interface. If it doesn't make sense to store this as an instance attribute, just remove the attribute and rely soley on the method parameter.

Reply:
Interesting.

Setting attributes that way seems error prone. If there is a typo in the YAML file parameter, such as `meeter_name`, it will silently set an incorrect attribute on the instance.

Reply:
@samdoran So, you are not wrong ; however, since this is an internal tool I think the potential risk for user errors due to incorrect naming conventions is acceptable to me. The Devs and QE set these values in the yaml to setup specific scenarios for testing purposes, so it will be our responsibility to fix these kind of mistakes anyway when the scenario we are trying to create doesn't function the way we expect.

Reply:
It looks like all of these values should be properties instead of local variables in this method. Maybe you could do just this one as a property to get things headed in the right direction. Maybe something like this.
```python
@property
def meter_id(self):
    if self._meter_id is None:
        self._meter_id = fake.uuid4()
    
    return self._meter_id

@property
def meter_name(self):
    if self._meter_name is None:
        service_tier, meter_sub, meter_name, units_of_measure = self._get_cached_meter_values(self.meter_id, self.SERVICE_METER)
        self._meter_name = meter_name
        
    return self._meter_name
```